docs/walkthrough.rst

```diff
@@ -0,0 +1,199 @@
+===========
+Walkthrough
+===========
+
+The aim of the four Annotation scripts is to provide a way to edit \
+Key-Value pairs (MapAnnotations) and Tags (TagAnnotations) by batch:
+
+* Import Key-Value pairs and Tags as Annotations from a .csv file
+* Export Key-Value pairs and Tags to a .csv file
+* Delete Key-Value pairs
+* Convert the namespaces of Key-Value pairs
+
+Following this walkthrough, we hope to give you an understanding of \
+the four scripts that will help you to manipulate your annotations.
+
+Object selection with the scripts
+---------------------------------
+
+The object selection logic is the same for all four scripts. In OMERO there are \
+two distinct hierarchies:
+
+* **Projects** → **Datasets** → **Images**
+* **Screens** → **Plates** → (**Wells** or **Acquisition/Run**) → **Images**
+
+More details about selecting High-Content-Screening are given in another \
+section (see :doc:`Selecting High-Content-Screening </extended_description>`).
+
+To select which objects to manipulate by batch, the scripts offer four distinct ways.
+
+Direct selection
+^^^^^^^^^^^^^^^^
+The first is to simply select the all the desired objects. Opening the script \
+after selecting the object will prefill the script parameters with the right \
+object type and IDs:
+
+.. figure:: images/annotation_01_selecting_all_autofill.png
+
+   *Selection of multiple datasets and the auto-filled Export to CSV script.*
+..
+
+For the direct selection, leave the Target Data Type as "*<selected>*" \
+(in this example, choosing **Dataset** would also work).
+
+Children selection
+^^^^^^^^^^^^^^^^^^
+Instead of selecting the objects one by one, we can select the parent object \
+and set the "*Target Data Type*" to the type of the children objects \
+we want to select.
+
+.. figure:: images/annotation_02_selecting_parentchild_autofill.png
+
+   *Selection of all the images of a project and the auto-filled Export to CSV script.*
+..
+
+In this example, we would select all the **Images** found under the **Project:701**.
+
+Tag selection
+^^^^^^^^^^^^^
+The third option is to select from a tag all the objects of a given type \
+(do not confiuse with the option of exporting tags). This grants additional  \
+flexibility, either to select objects attached to different projects \
+or different owners, or to have a finer control over \
+which objects to process.
+
+.. figure:: images/annotation_03_selecting_tags.png
+
+   *Selection of a subset of datasets from a tag and the auto-filled Export to CSV script.*
+..
+
+The selection works in the same way as for the children selection. Note \
+that choosing "*<selected>*" for "*Target Data Type*" will result in an \
+error in this case.
```

Comment:
Indeed it does. I found it the hard way. Yet again, after the users found it in another case during the workshop last week. (note, they really appreciated the power of these scripts in general.) In many cases, even in the first case above, it would make much more sense to pass by default "Image" as the Target Data Type -> this is what the user will want in vast majority of cases.

Reply:
I did the `<selected>` as default to give it the "what you select is what you get" behavior. 
-> I select dataset & click export right away, I get dataset annotations
-> I select images & click export right away, I get image annotations

I would also prefer not to change it because images are the last place I want to put annotations: I prioritize annotations on datasets or wells to avoid duplicating the same MapAnn on all images inside. 
This also comes from how I organize experiments: 
one sample/replicate -> one dataset. (always true for high-content-screening: one well -> one sample/replicate)

Reply:
> I did the `<selected>` as default to give it the "what you select is what you get" behavior.
> -> I select dataset & click export right away, I get dataset annotations
> -> I select images & click export right away, I get image annotations

sure, but as my comment indicates, one line in your list above is missing (that one which does not work so well) == ``-> I select Tag & click export right away and get an error`` - am I getting it right ? If yes, what is the reason to let me crash on a default Tag workflow ?
In the meantime, I understood your tactics on the 2 points you are listing and I can appreciate the logic there.

> how I organize experiments:

But the script is for general usership of OMERO ? Not for you solely ? :)

Reply:
>  I select Tag & click export right away and get an error

sorry I forgot to answer to that. That's indeed a shortcoming of my favorite default. And I don't have a way to change it in the UI to "Image" when the script is started from a tag. Maybe we have ways to address this later on (after merging the PR):
* implement the export of annotation on tags (and thus the list of tags and their IDs)
* have user preference in OMERO.web
* use caching to load in the UI the previous choices ?

> But the script is for general usership of OMERO ? Not for you solely ? :)

100% true, but for my defense, I also have users following my recommendation. That's also what I teach during my trainings. I guess you could call it the "Tom/Düsseldorf/GerBi annotation school" ;)

Reply:
- ``implement the export of annotation on tags (and thus the list of tags and their IDs)`` would be my preferred option

- All kudos for the Tom/.../GerBI annotation school way of thinking. But then, what is the way to achieve the diversity of annotations within images of the same Dataset ? Surely the annotations on Images have strong use-case ?